documentation/source/examples/analytical_rpcholesky.rst

```diff
@@ -0,0 +1,196 @@
+Analytical example with RPCholesky
+==================================
+
+In this example, we have data of:
+
+.. math::
+    x = \begin{pmatrix}
+        0.5 & 0.2 \\
+        0.4 & 0.6 \\
+        0.8 & 0.3
+    \end{pmatrix}
+
+We choose a ``SquaredExponentialKernel`` with ``length_scale`` of :math:`\frac{1}{\sqrt{2}}`, which produces the following Gram matrix:
+
+.. math::
+    \begin{pmatrix}
+        1.0 & 0.84366477 & 0.90483737 \\
+        0.84366477 & 1.0 & 0.7788007 \\
+        0.90483737 & 0.7788007 & 1.0
+    \end{pmatrix}
+
+Note that we do not need to precompute the full Gram matrix, the algorithm
+only needs to evaluate the pivot column at each iteration.
+
+The RPCholesky algorithm iteratively builds a coreset by:
+    - Sampling pivot points based on the residual diagonal of the kernel Gram matrix
+    - Updating an approximation matrix and the residual diagonal
+
+We ask for a coreset of size 2 in this example. We start with an empty coreset
+and an approximation matrix :math:`F = \mathbf{0}_{N \times k}`,
+where :math:`N = 3, k = 2` in our case.
+
+We first compute the diagonal of the Gram matrix as:
+
+.. math::
+    d = \begin{pmatrix}
+        1 \\
+        1 \\
+        1
+    \end{pmatrix}
+
+For the first iteration (i=0):
+
+1. We sample a pivot point proportional to their value on the diagonal. All choices are equally likely, so let us suppose we choose the pivot with index = 2.
+
+2. We now compute g, the column at index 2, as:
+
+.. math::
+    g = \begin{pmatrix}
+    0.90483737 \\
+    0.7788007 \\
+    1.0
+    \end{pmatrix}
+
+3. Remove overlap with previously chosen columns (not needed on the first iteration).
+
+4. Update the approximation matrix:
+
+.. math::
+    F[:, 0] = g / \sqrt{(g[2])} = \begin{pmatrix}
+    0.90483737 \\
+    0.7788007 \\
+    1.0
+    \end{pmatrix}
+
+5. Update the residual diagonal:
+
+.. math::
+    d = d - |F[:,0]|^2 = \begin{pmatrix}
+    0.18126933 \\
+    0.39346947 \\
+    0
+    \end{pmatrix}
+
+For the second iteration (i=1):
```

Comment:
I am finding it hard to follow, could you add more explanation? I am not sure if I understand it correctly: We start with gram-matrix, we make choice and we update the column of the matrix, then we update the diagonal and that is our new gram matrix?  If that is the case, could you add what the updated gram matrix looks like after the first iteration here? And then we look at the diagonal again?

Reply:
I hope the additional explanation of the algorithm in the beginning helps - let me know if it's still unclear